lib/shelley/exe/cardano-wallet-shelley.hs

```diff
@@ -0,0 +1,309 @@
+
+module Main where
+
+import Prelude
+
+import Cardano.BM.Data.Severity
+    ( Severity (..) )
+import Cardano.BM.Trace
+    ( Trace, appendName, logDebug, logError, logInfo, logNotice )
+import Cardano.CLI
+    ( LoggingOptions
+    , cli
+    , cmdAddress
+    , cmdByronWalletCreate
+    , cmdKey
+    , cmdMnemonicByron
+    , cmdNetwork
+    , cmdTransaction
+    , cmdVersion
+    , cmdWallet
+    , databaseOption
+    , enableWindowsANSI
+    , helperTracing
+    , hostPreferenceOption
+    , listenOption
+    , loggingMinSeverity
+    , loggingOptions
+    , loggingSeverityOrOffReader
+    , loggingTracers
+    , runCli
+    , setupDirectory
+    , shutdownHandlerFlag
+    , syncToleranceOption
+    , tlsOption
+    , withLogging
+    )
+import Cardano.Startup
+    ( ShutdownHandlerLog
+    , installSignalHandlers
+    , withShutdownHandler
+    , withUtf8Encoding
+    )
+import Cardano.Wallet.Api.Client
+    ( byronAddressClient
+    , byronTransactionClient
+    , byronWalletClient
+    , networkClient
+    )
+import Cardano.Wallet.Api.Server
+    ( HostPreference, Listen (..), TlsConfiguration )
+import Cardano.Wallet.Logging
+    ( trMessage, transformTextTrace )
+import Cardano.Wallet.Primitive.Types
+    ( SyncTolerance )
+import Cardano.Wallet.Shelley
+    ( TracerSeverities
+    , Tracers
+    , Tracers' (..)
+    , serveWallet
+    , setupTracers
+    , tracerDescriptions
+    , tracerLabels
+    )
+import Cardano.Wallet.Shelley.Launch
+    ( NetworkConfiguration (..)
+    , networkConfigurationOption
+    , nodeSocketOption
+    , parseGenesisData
+    )
+import Cardano.Wallet.Version
+    ( GitRevision, Version, gitRevision, showFullVersion, version )
+import Control.Applicative
+    ( Const (..), optional )
+import Control.Monad
+    ( void )
+import Control.Monad.Trans.Except
+    ( runExceptT )
+import Control.Tracer
+    ( contramap )
+import Data.Text
+    ( Text )
+import Data.Text.Class
+    ( ToText (..) )
+import Network.Socket
+    ( SockAddr )
+import Options.Applicative
+    ( CommandFields
+    , Mod
+    , Parser
+    , command
+    , helper
+    , info
+    , internal
+    , long
+    , metavar
+    , option
+    , progDesc
+    , value
+    )
+import System.Environment
+    ( getArgs, getExecutablePath )
+import System.Exit
+    ( ExitCode (..), exitWith )
+
+import qualified Data.Text as T
+import qualified Ouroboros.Consensus.Shelley.Protocol.Crypto as Shelley
+
+{-------------------------------------------------------------------------------
+                              Main entry point
+-------------------------------------------------------------------------------}
+
+main :: IO ()
+main = withUtf8Encoding $ do
+    enableWindowsANSI
+    runCli $ cli $ mempty
+        <> cmdServe
+        <> cmdMnemonicByron
+        <> cmdKey
+        <> cmdWallet cmdByronWalletCreate byronWalletClient
+        <> cmdAddress byronAddressClient
+        <> cmdTransaction byronTransactionClient byronWalletClient
```

Comment:
These will have to be change to point to the "normal" API. We could perhaps still have them behind a `legacy-wallet` and `legacy-address` sets of command.

Reply:
> These will have to be change to point to the "normal" API.

Done.